pinot-core/src/main/java/org/apache/pinot/core/query/reduce/GroupByDataTableReducer.java

```diff
@@ -289,6 +291,9 @@ public void runJob() {
 
               int numRows = dataTable.getNumberOfRows();
               for (int rowId = 0; rowId < numRows; rowId++) {
+                if (rowId % MAX_ROWS_UPSERT_PER_INTERRUPTION_CHECK == 0 && Thread.interrupted()) {
+                  throw new EarlyTerminationException();
```

Comment:
Question: Is it not enough to just `return` here? Given that the main thread already threw a timeout exception? It also keeps the behavior consistent with how we return for interrupts at L279.

Reply:
Good point, changed to return nothing.